### PR TITLE
repaths the demons

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

imps and slaughter demons shared 90% of their code and were in /simple_animal/ instead of simple_animal/hostile which meant they wouldnt attack mobs if spawned by admins or some other stuff
now imps are a child of hostile mobs and slaughter demons are a child of imps
removes unused code from imps
centers imp's left/right sprites because they were offcenter for some reason
changes the "slaughter" faction to "hell" on slaughter demons and adds the hell faction to bubblegum since they are all a part of the demon family and they shouldnt attack each other [refer to https://tgstation13.org/wiki/Slaughter_Demon]

## Why It's Good For The Game

some less shitcode and cool flavor, allows admins to spawn these mobs in and have them attack players

## Changelog
:cl:
tweak: the demonic hierarchy now shares the same factions
tweak: slaughter demons and imps spawned by admins or other sources that arent player controlled will attack people
code: slaughter demons and imps were repathed
/:cl:

### CODE REVIEW

**Comment by IndieanaJones** [comment]
I would've imagined that imps would be the subtype of slaughter demons, but I guess its fine either way.

**Comment by Fikou** [comment]
> I would've imagined that imps would be the subtype of slaughter demons, but I guess its fine either way.

this way is better because theres no snowflake code to change from slaughter demons

**Review by Twaticus — APPROVED** [review]

File: _maps/RandomZLevels/Academy.dmm [hunk 1]
```diff
@@ -2958,7 +2958,8 @@
 /turf/open/floor/plating,
 /area/awaymission/academy/academyengine)
 "lL" = (
-/mob/living/simple_animal/slaughter{
+/mob/living/simple_animal/hostile/imp/slaughter{
+	faction = list("hell", "neutral");
```
Comment (Rohesie) [inline _maps/RandomZLevels/Academy.dmm:2962]: Could you make a variant instead of var-editing it?

Comment (Fikou) [reply _maps/RandomZLevels/Academy.dmm:2962]: its a demon in an unaccessible part (behind unbreakable windows and walls) of a disabled away mission with 2 vars changed

Comment (Rohesie) [reply _maps/RandomZLevels/Academy.dmm:2962]:
```suggestion
/mob/living/simple_animal/hostile/imp/slaughter/demon_in_an_unaccessible_part_behind_unbreakable_windows_and_walls_of_a_disabled_away_mission_with_2_vars_changed
```

File: _maps/RandomZLevels/Academy.dmm [hunk 2]
```diff
@@ -2958,10 +2958,7 @@
 /turf/open/floor/plating,
 /area/awaymission/academy/academyengine)
 "lL" = (
-/mob/living/simple_animal/hostile/imp/slaughter{
-	faction = list("hell", "neutral");
-	name = "engine demon"
-	},
+/mob/living/simple_animal/hostile/imp/slaughter/demon_in_an_unaccessible_part_behind_unbreakable_windows_and_walls_of_a_disabled_away_mission_with_2_vars_changed,
```
Comment (Rohesie) [inline _maps/RandomZLevels/Academy.dmm:2961]:
```suggestion
/mob/living/simple_animal/hostile/imp/slaughter/engine_demon,
```

File: code/modules/antagonists/slaughter/slaughter.dm [hunk 3]
```diff
@@ -210,3 +210,7 @@
 		victim.forceMove(get_turf(victim))
 		victim.exit_blood_effect()
 		victim.visible_message("<span class='warning'>[victim] falls out of the air, covered in blood, looking highly confused. And dead.</span>")
+
+/mob/living/simple_animal/hostile/imp/slaughter/demon_in_an_unaccessible_part_behind_unbreakable_windows_and_walls_of_a_disabled_away_mission_with_2_vars_changed
```
Comment (Rohesie) [inline code/modules/antagonists/slaughter/slaughter.dm:214]:
```suggestion
/mob/living/simple_animal/hostile/imp/slaughter/engine_demon
```

**Pull request merged**